### ISSUE TITLE
(plugin typescript) SemiColons::Asi does not work properly

### ISSUE THREAD

**Issue by KSXGitHub** [dprint/dprint#197]
Full bug report is located at https://github.com/KSXGitHub/sane-fmt/tree/dprint-issue-semicolons-asi-does-not-work-properly (branch `dprint-issue-semicolons-asi-does-not-work-properly`)

**Comment by dsherret** [comment]
@KSXGitHub Thanks for the report! I will look at this tomorrow. It is a bit of extra work following this though especially on my phone. I’m also not going to run scripts on my machine because then I need to take the time to security audit them. In the future could you follow the bug report issue template (input code, expected output, actual output)? That will make it a little easier and outline everything here. Thanks!

**Comment by KSXGitHub** [comment]
@dsherret I see. The formatting rules is located at https://github.com/KSXGitHub/sane-fmt/blob/dprint-issue-semicolons-asi-does-not-work-properly/src/rules/mod.rs, and the file that needed formatting is https://github.com/KSXGitHub/sane-fmt/blob/dprint-issue-semicolons-asi-does-not-work-properly/scripts/sync-package-versions.js.

**Comment by KSXGitHub** [comment]
Also, I have created [a WASM program](https://github.com/KSXGitHub/sane-fmt/releases/tag/0.1.1), you may safely run it in a sandbox with wasmtime or wasmer.

**Comment by dsherret** [comment]
It's a lot of work to clone a repo and run things. This is much easier:

**Input Code**

```ts
for (const asdf of test) {
  const json = JSON.stringify(targetData, undefined, 2) + '\n'
}
```

**Expected Output**

```ts
for (const asdf of test) {
  const json = JSON.stringify(targetData, undefined, 2) + '\n'
}
```

**Actual Output**

```ts
for (const asdf of test) {
  const json = JSON.stringify(targetData, undefined, 2) + '\n';
}
```

**Comment by dsherret** [comment]
Fixed and released in 0.16.1. Had to do with some of my logic for variable declarations in for-like statements.
